### PR TITLE
Fixing a few node tests and triaging others

### PR DESCRIPTION
**Pull request by aseemw**
This PR fixes a few failing node tests and categorizes the rest so that it is easier to dedicate PRs to fix them in the future.

### CODE REVIEW

File: onnx_coreml/_backend.py [hunk 1]
```diff
@@ -3,11 +3,31 @@
...
+from typing import Any, Text, Dict
 from onnx import ModelProto
 from onnx.backend.base import Backend
 from onnx_coreml._backend_rep import CoreMLRep
 from onnx_coreml import convert
+import onnx
+from ._graph import _input_from_onnx_input
+
+
+def _get_onnx_outputs(model):
```
Comment (smessmer) [inline onnx_coreml/_backend.py]: Let's add type hints to all functions, without they're not type checked.

File: onnx_coreml/_backend.py [hunk 2]
```diff
@@ -20,7 +40,8 @@ def prepare(cls,
         # type: (...) -> CoreMLRep
         super(CoreMLBackend, cls).prepare(model, device, **kwargs)
         coreml_model = convert(model)
-        return CoreMLRep(coreml_model, device == 'CPU')
+        onnx_outputs = _get_onnx_outputs(model) # type: ignore
```
Comment (smessmer) [inline onnx_coreml/_backend.py]: You won't need `type: ignore` if `_get_onnx_outputs` has a type hint

File: onnx_coreml/_backend_rep.py [hunk 3]
```diff
@@ -7,11 +7,13 @@
 from typing import Any, Sequence, List
 from onnx.backend.base import BackendRep, namedtupledict
 from coremltools.models import MLModel  #type: ignore
+from typing import Dict, Any, Text
 
 
 class CoreMLRep(BackendRep):
     def __init__(self,
                  coreml_model,  # type: MLModel
+                 onnx_outputs,  # type: Dict[Text, Any]
```
Comment (smessmer) [inline onnx_coreml/_backend_rep.py]: Do we know a more concrete type than `Any` for an output? If it's a shape, something like `Tuple[int, ...]` might work.

File: onnx_coreml/converter.py [hunk 4]
```diff
@@ -23,37 +23,55 @@
...
+        else:
+            raise ValueError("Unrecognized input shape %s, for input '%s' " % (str(shape), str(input_[0])))
         features.append((str(input_[0]), datatypes.Array(*shape)))
     return features
 
+'''
+outputs: list of tuples. 
+      [Tuple]: [(name, type, shape)]
+'''
+def _make_coreml_output_features(outputs):  # type: (...) -> Sequence[Tuple[Text, datatypes.Array]]
```
Comment (smessmer) [inline onnx_coreml/converter.py]: There's a type hint for the parameter missing

File: onnx_coreml/_backend_rep.py (hunk 3 above)
Comment (aseemw) [reply onnx_coreml/_backend_rep.py]: Yes Tuple[int, ...] does work

File: onnx_coreml/_backend.py (hunk 2 above)
Comment (aseemw) [reply onnx_coreml/_backend.py]: I have removed the ignore.

**Pull request merged**